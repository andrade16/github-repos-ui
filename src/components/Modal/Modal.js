import React from "react";
import "./Modal.scss";

const Modal = ({ children, show, header, onClose }) => {
  return (
    <div className={`modal ${show ? "show" : ""}`} onClick={onClose}>
      <div
        className="modal-content"
        onClick={(event) => event.stopPropagation()}
      >
        <div className="modal-header">
          <h4 className="modal-title">{header}</h4>
        </div>
        <div className="modal-body">{children}</div>
        <div className="modal-footer">
          <button className="modal-button" onClick={onClose}>
            Close
          </button>
        </div>
      </div>
    </div>
  );
};

export default Modal;
