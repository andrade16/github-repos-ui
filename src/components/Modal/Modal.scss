@import "src/utils/variables";

.modal {
  position: fixed;
  left: 0;
  top: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(0, 0, 0, 0.5);
  display: flex;
  align-items: center;
  justify-content: center;
  opacity: 0;
  transition: all 0.3s ease-in-out;
  pointer-events: none;
  z-index: 2;
}

.modal.show {
  opacity: 1;
  pointer-events: visible;
}

.modal-content {
  display: flex;
  flex-direction: column;
  width: 500px;
  min-height: 230px;
  background-color: $white;
  transform: translateY(-200px);
  transition: all 0.3s ease-in-out;

  .modal-header,
  .modal-footer {
    padding: 10px;
  }

  .modal-header {
    border-bottom: 1px solid #eee;

    .modal-title {
      margin: 0;
    }
  }

  .modal-body {
    padding: 10px;
    overflow-y: scroll;
    max-height: 330px;
    flex: 1;
  }

  .modal-footer {
    width: 96%;
    border-top: 1px solid #eee;
    background-color: $white;

    .modal-button {
      color: $white;
      background-color: $red;
      border-color: $red;
      outline-color: $red;
      border-radius: $small-border-radius;
      height: 5vh;
      width: 10vw;
      min-width: 128px;
      cursor: pointer;
    }
  }
}

.modal.show .modal-content {
  transform: translateY(0);
}
