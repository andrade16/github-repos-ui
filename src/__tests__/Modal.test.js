import React from "react";
import renderer from "react-test-renderer";
import Modal from "../components/Modal/Modal";

describe("Modal", () => {
  it("Renders Modal correctly", () => {
    const tree = renderer
      .create(<Modal show={true} onClose={() => {}} header="Recent Commits" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
