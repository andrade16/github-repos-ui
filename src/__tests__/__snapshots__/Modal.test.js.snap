// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Renders Modal correctly 1`] = `
<div
  className="modal show"
  onClick={[Function]}
>
  <div
    className="modal-content"
    onClick={[Function]}
  >
    <div
      className="modal-header"
    >
      <h4
        className="modal-title"
      >
        Recent Commits
      </h4>
    </div>
    <div
      className="modal-body"
    />
    <div
      className="modal-footer"
    >
      <button
        className="modal-button"
        onClick={[Function]}
      >
        Close
      </button>
    </div>
  </div>
</div>
`;
